File: src/main/java/model/json/adapter/ContentAdapter.java
package model.json.adapter;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import java.nio.file.Path;
import java.nio.file.Paths;

public class ContentAdapter {

    private final CleanAdapter clean = new CleanAdapter();

    @ToJson String toJson(Path path) {
        return path.toString();
    }

    @FromJson
    Path fromJson(String path) {
        return Paths.get(clean.fromJson(path));
    }

}

File: src/main/java/model/json/adapter/ContentAdapter.kt
package model.json.adapter

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

import java.nio.file.Path
import java.nio.file.Paths

class ContentAdapter {

    private val clean = CleanAdapter()

    @ToJson
    internal fun toJson(path: Path): String = path.toString()

    @FromJson
    internal fun fromJson(path: String): Path = Paths.get(clean.fromJson(path))

}
